test/cppsuite/how_to_use_cppsuite.md

```diff
@@ -0,0 +1,175 @@
+We will cover in order the steps to [create your new test](#creating-a-new-test), [configure it](#configuring-tests), and finally [run the test](#running-tests).
+
+## Creating a new test
+New tests are created using the [create_test.sh](./create_test.sh) script in the root directory. This script takes the test name as its only argument, so we will call it with:  
+`./create_test.sh multi_size_inserts`  
+
+This will create two new files: the `.cpp` file `tests/multi_size_inserts.cpp` and the configuration file `configs/multi_size_inserts_default.txt`. Working with these files is covered below.  
+
+*Note: `s_all` is called as part of `create_test.sh` which automatically updates some boilerplate sections of the test framework based on the contents of `test_data.py`. If you want to change the name of an existing test you will need to also update it in `test_data.py` and then call `s_all` manually.*
+
+## Configuring tests
+Having generated these two files we now need to modify them to define our test.
+
+### The `.cpp` file
+
+The `.cpp` file allows users to define the behaviour of database operations for their test. This file is cloned from [test_template.cpp](tests/test_template.cpp) on creation and by default will do nothing as all functions are stubbed.
+
+#### Defining workloads
+In our example test `multi_size_inserts` we will first update the functions that define `insert` and `populate`.
+The default behavior of populate will create the table in the database, so we can delete the stubbed implementation of `populate` and make sure that it is defined in the [configuration file](#the-configuration-file) later.
+When cppsuite executes the test it will create threads that run the `insert_operation` function.  
+
+Initially, the insert function is stubbed as follows:
+```cpp
+void
+insert_operation(thread_worker *) override final
+{
+    logger::log_msg(LOG_WARN, "insert_operation: nothing done");
+}
+```
+The user can either define their own insertion logic by filling out the function body, or remove the `insert_operation` stub to fall back on the framework’s default implementation.
+For our `multi_size_inserts` example, we add code to to write transactions with either 10MB or 1KB updates. We will configure the `cache_size` to be 10MB later.
+
+```cpp
+void
+insert_operation(thread_worker *tw) override final
+{
+    // Retrieve the collection created by the populate operation.
+    collection &coll = tw->db.get_collection(0);
+    scoped_cursor cursor = tw->session.open_scoped_cursor(coll.name);
+
+    int value_size_bytes = 1000;
+
+    while (tw->running()) {
+
+        // Swap value size between 1KB and 10MB.
+        value_size_bytes = value_size_bytes == 1000 ? 10000000 : 1000;
```

Comment:
Isn't 1KB 1024 bytes, and 10MB 1024x1024 bytes?

Reply:
Updated in a1ee228bf1c057be8095493025bd6ae21c61d804

Reply:
Thanks.